src/main/java/org/stellar/sdk/Transaction.java

```diff
@@ -44,6 +32,7 @@ public class Transaction extends AbstractTransaction {
       Operation[] operations,
       Memo memo,
       TransactionPreconditions preconditions,
+      Optional<SorobanTransactionData> sorobanData,
```

Comment:
would discourage use of Optional for input parameters, it just results in more guard checking later, rather, just use class ref here, which null is acceptable.

Reply:
Fixed in 37e0e19accdadde748a715af4e5dbf25a73ad40e

Reply:
another good area to include in contributing code guidelines, https://github.com/stellar/java-stellar-sdk/issues/493